src/App.tsx

```diff
@@ -26,70 +28,125 @@ type ReorderOptions = {
 
-  return visibleGoods;
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const reorderedGoods = getReorderedGoods(goodsFromServer, this.state);
+    const isChanged = isReversed || sortType !== SortType.NONE;
+
+    const sortAlphabetically = () => {
+      this.setState({ sortType: SortType.ALPABET });
+    };
+
+    const sortByLength = () => {
+      this.setState({ sortType: SortType.LENGTH });
+    };
+
+    const reverse = () => {
+      this.setState(state => ({
+        isReversed: !state.isReversed,
+      }));
+    };
+
+    const reset = () => {
+      this.setState({
+        isReversed: false,
+        sortType: SortType.NONE,
+      });
+    };
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-info',
+              { 'is-light': sortType !== SortType.ALPABET },
+            )}
+            onClick={sortAlphabetically}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              { 'is-light': sortType !== SortType.LENGTH },
+            )}
+            onClick={sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-warning',
+              { 'is-light': !isReversed },
+            )}
+            onClick={reverse}
+          >
+            Reverse
+          </button>
+
+          {(isChanged) && (
```

Comment:
looks like those braces are unnecessary
```suggestion
          {isChanged && (
```